### DISCUSSION TITLE
How to set the status of an URL to "completed"?

### DISCUSSION THREAD

**Discussion by rzo1** [crawler-commons/url-frontier#46] [Q&A] [answered]
Probably a dumb question: How can I set the status of an URL to "completed", so it gets reported by GetStats as "completed", i.e. how to push custom counts to the url frontier? Is there an example available? 🙃

**Comment by jnioche** [comment, marked as answer]
This is a great question.

A URL counts as completed when it is received by the service, is not a _DiscoveredURLItem_ and does not have a _refetchable_from_date_ (or if its value is 0). If a URL has been fetched (i.e. it is not a _DiscoveredURLItem_) but is given a _refetchable_from_date_, it won't count towards the completed as it might be refetched later.

You can't push custom counts to the frontier. As you can see [in the code](https://github.com/crawler-commons/url-frontier/blob/master/service/src/main/java/crawlercommons/urlfrontier/service/AbstractFrontierService.java#L259) the _completed_ and _active_queues_ are not in the API and should be considered experimental and implementation dependent.

Does that answer your question @rzo1?

**Reply by rzo1** [reply]
Yes This answers the question. I could implement it accordingly.
